PublicMapping/districtbuilder
# Add email opt-in option and T&C to registration page

## Overview

This PR:
- adds `isMarketingEmailOn` field to data models: `IUser`, `User`, and `RegisterDto`
- adds a migration to add the above mentioned field to the `public.user` table
- changes method return shape per data model update
- updates the register endpoint async method
- updates the registration UI according to the design
- adds an email opt-in field and T&C text/link to the registration component, updates the form validator

### Checklist


### Demo

![Screen Shot 2022-05-06 at 5 54 03 PM](https://user-images.githubusercontent.com/16109558/167221616-0f1f5e3f-c9d1-4962-a98b-571f730cf62d.png)

## Testing Instructions

- Run migration
- Spin up servers and frontend
- Click on `Sign up for free` link on the home page
- Make sure the new email opt-in checkbox is there and is clickable
- Make sure the T&C text is there and the link can be clicked to open the T&C content in a new tab
- Fill in the form, make sure the form validator works as expected
- Submit the form. Open the dbshell and make sure the newly added field and value is correctly persisted

Closes #1123
